test/testvm.py

```diff
@@ -475,117 +604,42 @@ def __init__(self, **args):
-        gf.mkdir_p("/etc/systemd/system/sockets.target.wants/")
-        gf.ln_sf("/usr/lib/systemd/system/sshd.socket", "/etc/systemd/system/sockets.target.wants/")
-
-    def _setup_fedora_rawhide (self, gf):
-        self._setup_ssh_keys(gf)
-        self._setup_fedora_network(gf)
-
-    def _setup_rhel_7 (self, gf):
-        self._setup_ssh_keys(gf)
-        self._setup_fedora_network(gf)
-
-        # systemctl disable sshd.service
-        gf.rm_f("/etc/systemd/system/multi-user.target.wants/sshd.service")
-        # systemctl enable sshd.socket
-        gf.mkdir_p("/etc/systemd/system/sockets.target.wants/")
-        gf.ln_sf("/usr/lib/systemd/system/sshd.socket", "/etc/systemd/system/sockets.target.wants/")
-
-        # upload subscription information
-        gf.mkdir_p("/root/.rhel")
-        gf.upload(os.path.expanduser("~/.rhel/login"), "/root/.rhel/login")
-        gf.upload(os.path.expanduser("~/.rhel/pass"), "/root/.rhel/pass")
-
-    def run_modify_func(self, modify_func):
-        gf = guestfs.GuestFS(python_return_dict=True)
-        if self.verbose:
-            gf.set_trace(1)
-        try:
-            gf.add_drive_opts(self._image_image, readonly=False)
-            gf.launch()
-            # try to mount device directly
-            devices = gf.list_devices()
-            assert len(devices) == 1
-            try:
-                gf.mount(devices[0], "/")
-            except:
-                # if this fails, we may have to perform more intricate mounting
-                # get the first one that isn't swap and mount it as root
-                filesystems = gf.list_filesystems()
-                for fs in filesystems:
-                    if filesystems[fs] == "swap":
-                        continue
-                    gf.mount(fs, "/")
-                    if gf.exists("/etc"):
-                        break
-                    gf.umount("/")
-                if not gf.exists("/etc"):
-                    raise Failure("Can't find root partition")
-
-            modify_func(gf)
-            gf.touch("/.autorelabel")
-        finally:
-            gf.close()
+        self._fixed_mac_flavors = self._get_fixed_mac_flavors()
 
-    def unpack_base(self, modify_func=None):
-        assert not self._process
+        # it is ESSENTIAL to register the default implementation of the event loop before opening a connection
+        # otherwise messages may be delayed or lost
+        libvirt.virEventRegisterDefaultImpl()
+        self.virt_connection = libvirt.open("qemu:///session")
+        self.event_handler = VirtEventHandler(libvirt_connection=self.virt_connection, verbose=self.verbose)
 
-        if not os.path.exists(self.run_dir):
-            os.makedirs(self.run_dir, 0750)
+        # network names are currently hardcoded into network-cockpit.xml and network-cockpit_secondary.xml
+        self.network_name = self._read_network_name()
+        self.network_name_secondary = self._read_network_name_secondary()
```

Comment:
Same as above about secondary networks. I have a commit that removes them at stefwalter/test_libvirt.